app/controllers/pieces_controller.rb

```diff
@@ -19,7 +19,7 @@ def update
   private
   
     def set_piece
-      @piece = Piece.find(params[:id])
+      @piece ||= Piece.find(params[:id])
```

Comment:
I believe so. The method is called `set_piece`, which indicates it sets a property that's used later (rather than used directly). If it were called `piece` it'd make sense to use it in place of the `@piece` attribute (as a form of lazy loading).

IE: if this is what you're doing
```ruby
set_piece
something_with @piece
```
Then you shouldn't need those pipes `||=`
But if this is what you're doing:
```ruby
something_with set_piece
```
Then it should be called `piece` not `set_piece`, but the pipes can stay (then you're just using `@piece` as a cache).